internal/gatewayapi/runner/runner.go

```diff
@@ -69,11 +70,14 @@ func (r *Runner) subscribeAndTranslate(ctx context.Context) {
 		// gateway class linked to this controller
 		switch {
 		case gatewayClasses == nil:
+			// Envoy Gateway startup.
+			continue
+		case gatewayClasses[0] == nil:
 			// No need to translate, publish empty IRs, e.g. delete operation.
 			r.XdsIR.Delete(r.Name())
-			r.InfraIR.Delete(r.Name())
-		case gatewayClasses[0] == nil:
-			panic("gatewayclass is nil")
+			// A nil ProxyInfra tells the Infra Manager to delete the managed proxy infra.
```

Comment:
the comment says `nil` ProxyInfra but the code checks for `nil` Listeners
```
if in.Proxy.Listeners != nil {
```
I prefer having a `nil` Proxy and checking for that in the next stage

Reply:
Updated in commit 47fba0d.